File: src/main/java/wraith/fabricaeexnihilo/json/other/ColorJson.java
package wraith.fabricaeexnihilo.json.other;

import com.google.gson.*;
import wraith.fabricaeexnihilo.json.BaseJson;
import wraith.fabricaeexnihilo.util.Color;

import java.lang.reflect.Type;

public final class ColorJson extends BaseJson<Color> {

    private ColorJson() {}

    public static final ColorJson INSTANCE = new ColorJson();

    @Override
    public Color deserialize(JsonElement json, Type typeOfT, JsonDeserializationContext context) throws JsonParseException {
        return new Color(json.getAsString());
    }

    @Override
    public JsonElement serialize(Color src, Type typeOfSrc, JsonSerializationContext context) {
        return new JsonPrimitive(src.toHex());
    }


}

File: src/main/kotlin/exnihilofabrico/json/ColorJson.kt
package exnihilofabrico.json

import com.google.gson.*
import com.google.gson.reflect.TypeToken
import exnihilofabrico.util.Color
import java.lang.reflect.Type

object ColorJson: JsonDeserializer<Color>, JsonSerializer<Color> {
    val TYPE_TOKEN: Type = object: TypeToken<Color>(){}.type
    @Throws(JsonParseException::class)
    override fun deserialize(json: JsonElement, typeOfT: Type, context: JsonDeserializationContext) = Color(json.asString)

    @Throws(JsonParseException::class)
    override fun serialize(src: Color, typeOfSrc: Type, context: JsonSerializationContext) = JsonPrimitive(src.toHex())
}